code/processes/tailer.q

```diff
@@ -46,9 +47,40 @@ getprocs:{[x;y]
         }
 
 .servers.register[.servers.procstab;.tailer.tailreadertypes;1b]
+.servers.register[.servers.procstab;.tailer.eodtypes;1b]
 
 \d .
 
+/- eod - send end of day message to main EOD process
+endofday:{[pt;processdata]
+    .lg.o[`eod;"end of day message received - ",spt:string pt];
+
+    /- trigger final save down from tailer
+    endofdaysave[processdata];
+
+    /- find handle to send message to eod process
+    eodp:exec w from .servers.getservers[`proctype;.tailer.eodtypes;()!();1b;0b];
+
+    /- exit early if no eod process connected
+    if[0=count eodp;.lg.e[`connection;"no connection to the ",(string .tailer.eodtypes)," could be established, failed to send end of day message"];:()];
+
+    /- send procname to eod process so it loads correct TDB
+    procname:.proc.procname;
+    neg[first eodp](`endofday;pt;procname);
```

Comment:
Is this first in here just because eodp is a list? Am guessing only one eodprocess is connected at any point.

Reply:
Yeah you're right that's just because eodp is a list, when secondary eod sort processes do eventually get added they should only connect to the main eodprocess so there should only ever be one connected to the tailer